minimal-shadow-cljs-nodejs/src/server/main.cljs

```diff
@@ -2,210 +2,88 @@
-                                                                "sb-input1"
-                                                                "broadcast2:text-input"
-                                                                "value"]))
-                          (map :id (:slack/channels-raw @db)))
-        (callback nil #js {:statusCode 200}))))
-
+            [server.common :refer [clj->json decode-base64 parse-json json->clj]]
+            [server.slack.db :as slack-db]
+            [server.slack.handlers :as handlers]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;; SparkBoard SlackBot server
 
-(defn response [cb body]
-  (cb nil #js {:statusCode 200 :body body}))
+(defn body->clj [body]
+  (let [body (js->clj body :keywordize-keys true)]
+    (cond-> body
+      (:payload body)                                       ;; special case for Slack including a json-string as url-encoded body
+      (update :payload json->clj))))
 
-(defn handler [event _context callback]
+(j/defn handler*
   "Main AWS Lambda handler. Invoked by slackBot.
    See https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html"
-  (println "[handler] event:" event)
-  (let [body (j/get event :body)]
-    (println "[handler] body: " body)
-    (cond
-      ;; Slack API: identification challenge
-      (some #{"challenge"} (js-keys (parse-json body)))
-      (response callback (j/get (.parse js/JSON (j/get event :body)) :challenge))
-
-      ;; Slack Event triggered (e.g. global shortcut)
-      (:payload (uri/query-string->map (decode-base64 body)))
-      (handle-modal! callback (-> body
-                                  decode-base64
-                                  uri/query-string->map
-                                  :payload
-                                  parse-json))
-      
-      :else
-      (response callback (clj->js {:action "broadcast update request to project channels"
-                                   :channels (request-updates! (-> (.parse js/JSON body)
-                                                                   (j/get-in [:event :user])
-                                                                   slack-user
-                                                                   (get "name"))
-                                                               (map :id (:slack/channels-raw @db)))})))))
+  [^:js {:as req :keys [body]} ^js res]
+  (pp/pprint ["[handler] body:" body])
+
+  (p/try (p/let [response (cond
+                            ;; Slack API: identification challenge
+                            (:challenge body) (:challenge body)
+
+                            ;; Slack Interaction (e.g. global shortcut)
+                            (:payload body) (handlers/handle-interaction! (:payload body))
+
+                            ;; Slack Event
+                            (:event body)
+                            (handlers/handle-event! (:event body))
```

Comment:
Just daydreaming on API design for handlers: `handler/event!`?

Reply:
Sounds good